src/gui/widgets/SpinPageAdapter.cpp

```diff
@@ -37,7 +34,16 @@ void SpinPageAdapter::pageNrSpinChangedCallback(GtkSpinButton* spinbutton, SpinP
     adapter->lastTimeoutId = g_timeout_add(100, reinterpret_cast<GSourceFunc>(pageNrSpinChangedTimerCallback), adapter);
 }
 
-auto SpinPageAdapter::getWidget() -> GtkWidget* { return this->widget; }
+auto SpinPageAdapter::newWidget() -> GtkWidget* {
+    g_object_unref(this->widget);
```

Comment:
Not quite happy with this. This method does not make it obvious that through the creation of a new widget the old one is dropped into a partially unconnected state. Also the widget is in this case the listener and should therefore be connected to the adapter. The adapter should have nothing to do with creating the widget. Therefore create the widget in the ToolPageSpinner itself and then attach it to the SpinPageAdapter. For this you can create a new interface for connecting widgets to this adapter. Then either make it obvious that the adapter can only handle one attached widget or allow for multiple widgets to be attached.